client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -217,6 +217,22 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionPool con
         return createRevisionedStreamClient(getSegmentForRevisionedClient(scope, streamName), serializer, config);
     }
 
+    @Override
+    public <T> RevisionedStreamClient<T> createRevisionedStreamClient(String streamName, Controller controllerObj,
+                                                                      Serializer<T> serializer,
+                                                                      SynchronizerConfig config,
+                                                                      AccessOperation accessOperation) {
+        log.info("Creating revisioned stream client for stream {} with synchronizer configuration {} and access operation: {}",
```

Comment:
can we reuse and consolidate the logic in this and its namesake method. a lot of it seems common except the instantiation of controller from client config.

Reply:
Sure, I've made the changes as suggested.